frontend-react/src/components/ManagePublicKey/ManagePublicKey.tsx

```diff
@@ -82,50 +91,77 @@ function ManagePublicKeySwitchDisplay() {
         }
     };
 
+    const showPublicKeyConfigured =
+        sender.length > 0 &&
+        hasPublicKey &&
+        !uploadNewPublicKey &&
+        !fileSubmitted;
+    const showUploadMsg =
+        (sender.length > 0 && !fileSubmitted && !hasPublicKey) ||
+        (!uploadNewPublicKey && sender.length === 0);
+    const showUpload =
+        (sender.length > 0 && !fileSubmitted && !hasPublicKey) ||
+        uploadNewPublicKey;
+    const showUploadError = fileSubmitted && !isUploading && !isSuccess;
```

Comment:
You should format the boolean names here to be is/hasX instead of showX. Also, an empty string is considered falsey and can be used directly in boolean rules.